pymeasure/instruments/keysight/keysightN7744C.py

```diff
@@ -0,0 +1,158 @@
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+from pymeasure.instruments import Instrument, SCPIMixin, Channel
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class MPPMChannel(Channel):
+    """A channel of the voltage source."""
+
+    wavelength = Channel.control(
+        "SENSe{ch}:POWer:WAVelength?", "SENSe{ch}:POWer:WAVelength %gNM",
+        """Control the sensor wavelength of this channel.""",
+        validator=strict_range,
+        values=[1250, 1650],
+        get_process=lambda v: v / 1e-9
+    )
+
+    power = Channel.measurement(
+        "READ{ch}:POWer?",
+        """Measure the current power meter value of this channel.""",
+    )
+
+    power_unit = Channel.control(
+        "SENSe{ch}:POWer:UNIT?", "SENSe{ch}:POWer:UNIT %g",
+        """Control the sensor wavelength of this channel.""",
+        validator=strict_discrete_set,
+        map_values=True,
+        values={'dBm': 0, 'Watt': 1}
+    )
+
+    range = Channel.control(
+        "SENSe{ch}:POWer:RANGe?", "SENSe{ch}:POWer:RANGe %g",
+        """Control the range setting for the channel.""",
+        validator=strict_range,
+        values=[-30, 10]
+    )
+
+    auto_range = Channel.control(
+        "SENSe{ch}:POWer:RANGe:AUTO?", "SENSe{ch}:POWer:RANGe:AUTO %g",
+        """Control whether automatic power ranging is being used by the channel.""",
+        validator=strict_discrete_set,
+        map_values=True,
+        values={False: 0, True: 1}
+    )
+
+    averaging_time = Channel.control(
+        "SENSe{ch}:POWer:ATIMe?", "SENSe{ch}:POWer:ATIMe %g",
+        """Control the averaging time of the channel.""",
+    )
+
+    auto_gain = Channel.control(
+        "SENSe{ch}:POWer:GAIN:AUTO?", "SENSe{ch}:POWer:GAIN:AUTO %g",
+        """Control the Auto Gain of the channel.""",
+        validator=strict_discrete_set,
+        map_values=True,
+        values={False: 0, True: 1}
+    )
+
+    zero = Channel.control(
+        "SENSe{ch}:CORRection:COLLect:ZERO?", "SENSe{ch}:CORRection:COLLect:ZERO",
+        """Control the electrical offsets of the channel.""",
+    )
+
+    state = Channel.control(
+        "SENSe{ch}:FUNCtion:STATe?", "SENSe{ch}:FUNCtion:STATe %s",
+        """Control the logging, MinMax, or stability data acquisition function mode.""",
+        validator=strict_discrete_set,
+        values=['LOGGing', 'LOGG', 'STABility', 'STAB', 'MINMax', 'MINM', 'STOP', 'START']
+    )
+
+    logging_parameters = Channel.control(
+        "SENSe{ch}:FUNCtion:PARameter:LOGGing?",
+        "SENSe{ch}:FUNCtion:PARameter:LOGGing %g,%gS",
+        """Control the number of data points and the averaging time for the logging data
+        acquisition function."""
+        # get_process=lambda v: tuple(float(x) for x in v.split(','))
```

Comment:
If it is not needed anymore, please remove it.
It seems quite the standard response processing.